force-app/main/default/classes/ExtensionControllerTest.cls

```diff
@@ -0,0 +1,142 @@
+@isTest
+private class ExtensionControllerTest {
+    public String searchText {get;set;}
+	private static void initData(String actionToDo) {
+        switch on actionToDo {
+            when 'testMethodNameSuccess' {
+                
+                //Dummy Account for test
+                Account acct = new Account();
+                acct.Name = 'TestAccount';
+                insert acct;
+                
+                //Realted dummy Contacts for test
+                List<Contact> contacts = new List<Contact>();
+                Contact testCon1 = new Contact(LastName = 'TestCon1', AccountId = acct.Id);
+                Contact testCon2 = new Contact(LastName = 'TestCon2', AccountId = acct.Id);
+                Contact testCon3 = new Contact(LastName = 'TestCon3', AccountId = acct.Id);
+                Contact testCon4 = new Contact(LastName = 'TestCon4', AccountId = acct.Id);
+                contacts.add(testCon1);
+                contacts.add(testCon2);
+                contacts.add(testCon3);
+                contacts.add(testCon4);
+                insert contacts;
+               
+            }
+         
+            when 'testMethodNameFailure' {
+                
+                //Dummy account to use for failing test
+                Account testacct = new Account(Name='FailAcc');
+                insert testacct;
+			}
+        }
+}
+    
+    //Testing setPrimary() method
+    @isTest
+    private static void testsetPrimarySuccess() {
+        initData('testMethodNameSuccess');
+        
+         Account refAcct = [SELECT Id,Name FROM Account WHERE Name = 'TestAccount'];
```

Comment:
you can create private static variable and assign refAcct in initData
The same for refCon